client/sidebar/footer/voip/index.tsx

```diff
@@ -1,17 +1,41 @@
+import { Random } from 'meteor/random';
 import React, { ReactElement, useCallback, useState } from 'react';
 
+import { roomTypes } from '../../../../app/utils/client';
 import { useCallActions, useCallerInfo } from '../../../contexts/CallContext';
+import { useEndpoint } from '../../../contexts/ServerContext';
 import { useTranslation } from '../../../contexts/TranslationContext';
+import { useUser } from '../../../contexts/UserContext';
 import { VoipFooter as VoipFooterComponent } from './VoipFooter';
 
 export const VoipFooter = (): ReactElement | null => {
 	const t = useTranslation();
 	const callerInfo = useCallerInfo();
 	const callActions = useCallActions();
+	const user = useUser();
 
 	const [muted, setMuted] = useState(false);
 	const [paused, setPaused] = useState(false);
 
+	const visitorEndpoint = useEndpoint('POST', 'livechat/visitor');
+	const voipEndpoint = useEndpoint('GET', 'voip/room');
+
+	const openRoom = useCallback(async () => {
```

Comment:
How do you feel about moving the function to the voice context and only calling it here?